src/clib/pioc_support.c

```diff
@@ -148,35 +188,43 @@ void pio_log(int severity, const char *fmt, ...)
        many tabs before the message. */
     if (!severity)
     {
-        strcpy(ptr, ERROR_PREFIX);
+        strncpy(ptr, ERROR_PREFIX, (rem_len > 0) ? rem_len : 0);
         ptr += strlen(ERROR_PREFIX);
+        rem_len -= strlen(ERROR_PREFIX);
     }
     for (t = 0; t < severity; t++)
-        strcpy(ptr++, "\t");
+    {
+        strncpy(ptr++, "\t", (rem_len > 0) ? rem_len : 0);
+        rem_len--;
+    }
 
     /* Show the rank. */
-    sprintf(rank_str, "%d ", my_rank);
-    strcpy(ptr, rank_str);
+    snprintf(rank_str, MAX_RANK_STR, "%d ", my_rank);
+    strncpy(ptr, rank_str, (rem_len > 0) ? rem_len : 0);
```

Comment:
There are a bunch of copies to the same buffer. Having a consistent style of checking the remaining length is easier to read.